docs/pages/router/create-pages.mdx

```diff
@@ -112,3 +112,36 @@ export default function Page() {
   return <Text>Blog post: {slug}</Text>;
 }
 ```
+
+## Non-route files
+
+Every file and folder in the `app/` folder is either a `_layout` file, or a route in your app. This means that other files like your components, hooks, utilities etc cannot be placed in the `app/` folder because they're neither screens nor layout files.
+
+<FileTree
+  files={[
+    'app/(home)/_layout.tsx',
+    ['app/(home)/profile.tsx', '/profile'],
+    'app/(modal)/_layout.tsx',
+    ['app/(modal)/profile.tsx', '/profile'],
+  ]}
+/>
+
+The example above demonstrates why, if Expo Router allowed for non-route files, co-location (the technical term for keeping files within the same directory) would lead to an unstructured project. [Shared routes](/router/advanced/shared-routes) (the feature allowing you render the same route with different layouts) is common pattern in native navigation, but requires multiple directories. As you will have multiple route files, but only a single file for components co-location would force you to create arbitrary rules over which directory is the 'correct' one. Expo Router does not allow for co-location preventing this dilemma.
+
+export const ExampleStructure = () => (
+  <FileTree
+    files={[
+      'app/sign-in',
+      'app/sign-out',
+      'app/profile',
+      'app/tasks',
+      'components/authentication',
+      'components/tasks/overview',
+      'components/profile/[user]',
+    ]}
+  />
+);
+
+> **info** **Tip**: Expo recommends sorting components and hooks by "feature" and routes by "navigation pattern". While the two concepts can overlap, there is enough difference to structure them differently. Additionally routes will inevitably change, but components generally stay within the same feature set. Keeping routes separate will greatly simplify your refactors.<br /><br /> <ExampleStructure />
+> In the example above both `/sign-in` and `/sign-out` are authentication routes and share components from `components/authentication`. However `/profile/[user]` is complex and renders components that belong to the authentication, profile and task features. If we place our components within the `app/` directory, its not obvious where these components belong. By separating components from routes you are creating an organized hierarchy with the philosophy that a route is a collection of reusable components, yet a component does not belong to an individual route. <br /><br />
+> Another project optimization is to utilize [path aliases](/guides/typescript/#path-aliases-optional). Path aliases keep your routes project structure agnostic, as imports within your routes are no longer dependent on our `app/` directory structure. This avoids large `import` statement changes when adding new directories (such as new groups) and prevents lengthy `import` paths.
```

Comment:
Probably don't need all of these to be in call-out blocks.

Reply:
I agree with Evan's comment above. This should be part of the main section or a sub-section to "Non-route files".